src/components/expression/expression-renderer.js

```diff
@@ -968,13 +1118,40 @@ genotet.ExpressionRenderer.prototype.highlightLabelsAfterUpdateData_ =
       this.clickedObject_.column = heatmapData.conditionNames.indexOf(
         this.clickedObject_.conditionName
       );
-      this.signal('expressionClick', this.clickedObject_);
+    }
+    if (this.clickedObject_.row != -1 && this.clickedObject_.column != -1) {
+        this.signal('expressionClick', this.clickedObject_);
     } else {
       this.signal('expressionUnclick');
     }
   };
 
 /**
+ * Load expression matrix data after zoom in and out the heatmap.
+ * @param {!Object} params Parameters for expression zoom in.
```

Comment:
Please be specific about Object. It is encouraged to write object attributes explicitly.